request.js

```diff
@@ -390,12 +390,16 @@ Request.prototype.init = function (options) {
     if (Object.prototype.hasOwnProperty.call(options.auth, 'password')) {
       options.auth.pass = options.auth.password
     }
+    if (Object.prototype.hasOwnProperty.call(options.auth, 'negotiate')) {
```

Comment:
On 302 redirects, we never get back into the ```if (options.auth) {``` block to re-negotiate on the redirected path.